bookkeeper-common-allocator/src/main/java/org/apache/bookkeeper/common/allocator/LeakDetectionPolicy.java

```diff
@@ -43,5 +43,15 @@ public enum LeakDetectionPolicy {
      * stack traces of places where the buffer was used. Introduce very
      * significant overhead.
      */
-    Paranoid,
+    Paranoid;
+
+    public static LeakDetectionPolicy parseLevel(String levelStr) {
+        String trimmedLevelStr = levelStr.trim();
+        for (LeakDetectionPolicy policy : values()) {
+            if (trimmedLevelStr.equalsIgnoreCase(policy.name())) {
+                return policy;
+            }
+        }
+        return LeakDetectionPolicy.Disabled;
```

Comment:
Add warn log when parse level string failed.

Reply:
We'd better print out the `levelStr` in the log. Maybe we need to change the log level to ERROR instead of WARN?

Reply:
> We'd better print out the levelStr in the log

Make sense

> Maybe we need to change the log level to ERROR instead of WARN?

I think warn is enough. It can work well. If we throw an exception, we can use ERROR log.

Reply:
Use: `log.warn("Parse leak detection policy level {} failed. Use the default level: {}", levelStr, LeakDetectionPolicy.Disabled.name());`